plugins/defaultplugins/ifplugin/vppdump/dump_nat_vppcalls.go

```diff
@@ -0,0 +1,463 @@
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/ifplugin/vppcalls"
+)
+
+// Nat44GlobalConfigDump returns global config in NB format
+func Nat44GlobalConfigDump(swIfIndices ifaceidx.SwIfIndex, log logging.Logger, vppChan *govppapi.Channel, stopwatch *measure.Stopwatch) (*nat.Nat44Global, error) {
+	// Dump all necessary data to reconstruct global NAT configuration
+	isEnabled, err := nat44IsForwardingEnabled(log, vppChan, measure.GetTimeLog(&bin_api.Nat44ForwardingIsEnabled{}, stopwatch))
+	if err != nil {
+		return nil, err
+	}
+	natInterfaces, err := nat44InterfaceDump(swIfIndices, log, vppChan, measure.GetTimeLog(&bin_api.Nat44InterfaceDump{}, stopwatch))
+	if err != nil {
+		return nil, err
+	}
+	natOutputFeature, err := nat44InterfaceOutputFeatureDump(swIfIndices, log, vppChan, measure.GetTimeLog(&bin_api.Nat44InterfaceDump{}, stopwatch))
+	if err != nil {
+		return nil, err
+	}
+	natAddressPools, err := nat44AddressDump(log, vppChan, measure.GetTimeLog(&bin_api.Nat44AddressDump{}, stopwatch))
+	if err != nil {
+		return nil, err
+	}
+
+	// Combine interfaces with output feature with the rest of them
+	var nat44GlobalInterfaces []*nat.Nat44Global_NatInterfaces
+	for _, natInterface := range natInterfaces {
+		nat44GlobalInterfaces = append(nat44GlobalInterfaces, &nat.Nat44Global_NatInterfaces{
+			Name:     natInterface.Name,
+			IsInside: natInterface.IsInside,
+			OutputFeature: func(ofIfs []*nat.Nat44Global_NatInterfaces, ifName string) bool {
+				for _, ofIf := range ofIfs {
+					if ofIf.Name == ifName {
+						return true
+					}
+				}
+				return false
+			}(natOutputFeature, natInterface.Name),
+		})
+	}
+
+	// Set fields
+	return &nat.Nat44Global{
+		Forwarding:    isEnabled,
+		NatInterfaces: nat44GlobalInterfaces,
+		AddressPools:  natAddressPools,
+	}, nil
+}
+
+func NAT44DNatDump(swIfIndices ifaceidx.SwIfIndex, log logging.Logger, vppChan *govppapi.Channel, stopwatch *measure.Stopwatch) (*nat.Nat44DNat, error) {
+	// Dump all necessary data to reconstruct DNAT configuration
+	natStMappings, err := nat44StaticMappingDump(swIfIndices, log, vppChan, measure.GetTimeLog(&bin_api.Nat44StaticMappingDump{}, stopwatch))
+	if err != nil {
+		return nil, err
+	}
+	natStLbMapping, err := nat44StaticMappingLbDump(log, vppChan, measure.GetTimeLog(&bin_api.Nat44LbStaticMappingDump{}, stopwatch))
+	if err != nil {
+		return nil, err
+	}
+	natIdMappings, err := nat44IdentityMappingDump(swIfIndices, log, vppChan, measure.GetTimeLog(&bin_api.Nat44IdentityMappingDump{}, stopwatch))
+	if err != nil {
+		return nil, err
+	}
+
+	// Append static mappings
+	natStMappings = append(natStMappings, natStLbMapping...)
+
+	// Reconstruct DNat config object
+	var dNatConfigs []*nat.Nat44DNat_DNatConfig
+	dNatConfig := &nat.Nat44DNat_DNatConfig{
+		StMappings: natStMappings,
+		IdMappings: natIdMappings,
+	}
+	// Currently it is not possible to distinguish which mapping belongs to which DNAT configuration, so everything
+	// is returned as a one config
```

Comment:
Would it be possible to update VPP to include this commit:
```
commit 5f22499e8836066a268dcb3f4d16cfcd0244b2a2
Author: Matus Fabian <matfabia@cisco.com>
Date:   Thu Jan 25 21:59:16 2018 -0800

    NAT44: add opaque string tag to static mapping APIs (VPP-1147)
    
    Change-Id: I620e2081285ca8ac5c2da8efc12fe6f540ea4fd1
    Signed-off-by: Matus Fabian <matfabia@cisco.com>
```
The tag allows to connect mappings with the DNatConfig.
In contiv/vpp we need to get the same data through the dumps that were previously passed in via the northbound API.

Reply:
The VPP update + change for the DNatConfig dump will be done in a separate pull request.